packages/merge-on-green/src/merge-logic.ts

```diff
@@ -0,0 +1,537 @@
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<ReviewsRequested> {
+  try {
+    return (
+      await github.pulls.listReviewRequests({
+        owner,
+        repo,
+        pull_number: pr,
+      })
+    ).data as ReviewsRequested;
+  } catch (err) {
+    return { users: [], teams: [] };
+  }
+};
+
+//this function cleans the reviews, since the listReviews method github provides returns a complete history of all comments added
+//and we just want the most recent for each reviewer
+mergeOnGreen.cleanReviews = function cleanReviews(
+  reviewsCompleted: Reviews[]
+): Reviews[] {
+  const cleanReviews = [];
+  const distinctReviewers: string[] = [];
+  for (let x = reviewsCompleted.length - 1; x >= 0; x--) {
+    const reviewsCompletedUser = reviewsCompleted[x].user.login;
+    if (!distinctReviewers.includes(reviewsCompletedUser)) {
+      cleanReviews.push(reviewsCompleted[x]);
+      distinctReviewers.push(reviewsCompletedUser);
+    }
+  }
+  return cleanReviews;
+};
+
+// TODO: test the logic for a team that's been assigned to review.
+// we might also want to make whether or not a requested viewer needs
+// to have been reviewed.
+mergeOnGreen.checkReviews = async function checkReviews(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+): Promise<boolean> {
+  console.info('=== checking required reviews ===');
+  let reviewsPassed = true;
+  const reviewsCompletedDirty = await mergeOnGreen.getReviewsCompleted(
+    owner,
+    repo,
+    pr,
+    github
+  );
+  const reviewsRequested = await mergeOnGreen.getReviewsRequested(
+    owner,
+    repo,
+    pr,
+    github
+  );
+  if (reviewsCompletedDirty.length !== 0) {
+    const reviewsCompleted = mergeOnGreen.cleanReviews(reviewsCompletedDirty);
+    if (reviewsCompleted.length !== 0) {
+      reviewsCompleted.forEach(review => {
+        if (review.state !== 'APPROVED') {
+          console.log('One of your reviewers did not approve the PR');
+          reviewsPassed = false;
+        }
+      });
+    }
+  } else {
+    //if no one has reviewed it, fail the merge
+    console.log('No one has reviewed your PR');
+    reviewsPassed = false;
+    return reviewsPassed;
+  }
+  if (
+    reviewsRequested.users.length !== 0 ||
+    reviewsRequested.teams.length !== 0
+  ) {
+    console.log('You have assigned reviewers that have not submitted a PR');
+    reviewsPassed = false;
+    return reviewsPassed;
+  }
+  return reviewsPassed;
+};
+
+mergeOnGreen.merge = async function merge(
+  owner: string,
+  repo: string,
+  pr: number,
+  github: GitHubAPI
+) {
+  const commitInfo = await mergeOnGreen.getPR(owner, repo, pr, github);
+  try {
+    const merge = await github.pulls.merge({
+      owner,
+      repo,
+      pull_number: pr,
+      commit_title: commitInfo.title,
+      commit_message: commitInfo.body,
+      merge_method: 'squash',
+    });
+    return merge;
+  } catch (err) {
```

Comment:
let's not have a `try`/`catch` here, I noticed earlier today testing that if an error occurs we delete the tracking entry from the database, because the behavior is the same as a success. we should instead print an error.